### ISSUE TITLE
Unidentified index. Name, surname, in Wallboard.php

### ISSUE THREAD

**Issue by Lurch440** [zbxwallboard/ZbxWallboard#4]
Hi, 

This kind of error.  Other it works just fine.

![Screenshot_2019-03-27 IT Environment Health](https://user-images.githubusercontent.com/11337811/55074566-4dfaa100-5099-11e9-8f35-8036991ef7ba.png)

Zabbix version 4.0
The user for Wallboard is in the guest group.  Guest are able to view just a few host groups.
![Screenshot_2019-03-27 Configuration of user groups](https://user-images.githubusercontent.com/11337811/55074897-307a0700-509a-11e9-8cfa-3cd6aff529b2.png)

When I login (top-right) of the Wallboard, I am able to add acknowledges, and there are no error message.
![Screenshot_2019-03-27 IT Environment Health(1)](https://user-images.githubusercontent.com/11337811/55075022-864eaf00-509a-11e9-8f97-9facf86a8ee9.png)

**Comment by mjaepel** [comment]
Seems that your user have empty name/surname in Zabbix. You could set this as workaround and the warning will disappear.
But you are right the wallboard should check this also and display the username instead.

**Comment by Lurch440** [comment]
That's the weird thing.  Names are given for each user.
![Screenshot_2019-03-27 Configuration of users](https://user-images.githubusercontent.com/11337811/55083752-b0a96800-50ac-11e9-9cb8-7420bffd8aba.png)

**Comment by Lurch440** [comment]
... but even when I am signed into Wallboard as myself, and there are no error message, the comment still doesn't show my name.

**Comment by mjaepel** [comment]
I tested it with Zabbix 4.0.5 again and I am unable to reproduce it.
Name is always displayed if set on user in Zabbix. If not then the PHP warnings occur.
So I could fix the PHP warnings but I have no idea why the names are missing in your setup.

Which Zabbix auth backend do you use? intern, ldap or http?
Is the user information for acknowlegement in Zabbix frontend correctly shown?

Maybe you could add a debug output to
classes/Wallboard.php - Line 269
```
	var_dump($EVENT);
	foreach ($EVENT[0]["acknowledges"] as $ACKED_MSG) {
```

You will see the raw data in acknowledge details overlay. There should no sensitive data - please copy the text.

**Comment by Lurch440** [comment]
Ahhh... am using LDAP.
So, it’s probably looking not looking for givenName or SN.

I’ll make a local user a test with that.

Log to follow when i get to the office.


Cheers,

Alex

[cid:part1.A23C47F4.E631EAC3@metcalfe.fi]+358 (0)404 160 190<tel:+358%20404%20160%20190>

[cid:part2.3968616F.1ACEA6E7@metcalfe.fi]https://telegram.me/aksu358

[cid:part4.A94BCD95.50FA5DBA@metcalfe.fi] alexmetcalfe.blogsite.org<http://alexmetcalfe.blogsite.org/>

[cid:part6.1518F622.CDDF0E56@metcalfe.fi]aksu358<callto://aksu358>

On 28 Mar 2019, at 23.09, Firewire2002 <notifications@github.com<mailto:notifications@github.com>> wrote:


I tested it with Zabbix 4.0.5 again and I am unable to reproduce it.
Name is always displayed if set on user in Zabbix. If not then the PHP warnings occur.
So I could fix the PHP warnings but I have no idea why the names are missing in your setup.

Which Zabbix auth backend do you use? intern, ldap or http?
Is the user information for acknowlegement in Zabbix frontend correctly shown?

Maybe you could add a debug output to
classes/Wallboard.php - Line 269

        var_dump($EVENT);
        foreach ($EVENT[0]["acknowledges"] as $ACKED_MSG) {


You will see the raw data in acknowledge details overlay. There should no sensitive data - please copy the text.

**Comment by mjaepel** [comment]
It's not related to auth backend. It's easier:
The user which is used for wallboard is not allowed to read user information. "Zabbix User" and "Zabbix Admin" could only see their own user details but not of other users. Only "Zabbix SuperAdmin" is allowed to read it.

So I could only display what I get from Zabbix API. ;)

**Comment by mjaepel** [comment]
It's the same in Zabbix frontend:

![Inaccessible_user](https://user-images.githubusercontent.com/7748344/55255599-526eb780-525b-11e9-8b9f-16078a03fb1e.png)

**Comment by mjaepel** [comment]
Fixed in a8ddeb6c857dbccc38c3c8db67655e1b61083ec5 and Release [v2019-03-29](https://github.com/zbxwallboard/ZbxWallboard/releases/tag/2019-03-29)
